### PR TITLE
Feat: Add manual assert 

### LINKED ISSUE #62: Add `manual_assert`
# What it does

Detects if-then-panic! that can be replaced with assert!.

# Why is this bad?

`assert!` is simpler than if-then-panic!.

# Example

```cairo
let sad_people: Array<&str> = arrray![];
if !sad_people.is_empty() {
    panic!("there are sad people: {:?}", sad_people);
}
```

Use instead:

```cairo
let sad_people: Array<&str> = array![];
assert!(sad_people.is_empty(), "there are sad people: {:?}", sad_people);
```
https://rust-lang.github.io/rust-clippy/master/index.html#/manual_assert

### PR DESCRIPTION
**Pull request by stevencartavia**
- Resolves #62 
- Detects if-then-panic! that can be replaced with assert!.

### CODE REVIEW

File: crates/cairo-lint-core/src/fix.rs [hunk 1]
```diff
@@ -378,4 +379,44 @@ impl Fixer {
...
+        };
+
+        let block_expr = if_expr.if_block(db);
+        let statements = block_expr.statements(db).elements(db);
+
+        if let Some(Statement::Expr(statement_expr)) = statements.first() {
+            if let Expr::InlineMacro(inline_macro) = &statement_expr.expr(db) {
+                let panic_message = inline_macro.arguments(db).as_syntax_node().get_text_without_trivia(db);
+                let condition_text = condition.as_syntax_node().get_text_without_trivia(db);
+                let assert_condition = condition_text.trim_start_matches('!');
```
Comment (0xLucqs) [inline crates/cairo-lint-core/src/fix.rs:398]: don't do that, take the if condition from the `Expr::If`

**Review by 0xLucqs — CHANGES_REQUESTED** [review]

File: crates/cairo-lint-core/src/fix.rs [hunk 2]
```diff
@@ -502,4 +503,31 @@ impl Fixer {
...
+                let panic_message = inline_macro.arguments(db).as_syntax_node().get_text_without_trivia(db);
+
+                let indentation = node.get_text(db).chars().take_while(|c| c.is_whitespace()).collect::<String>();
+
+                let assert_macro = format!(
+                    "{}assert!({}, \"{}: {{:?}}\", {});\n",
+                    indentation,
+                    condition_text,
+                    panic_message.trim_matches(&['(', ')'][..]).replace("\"", "").trim(),
+                    condition_text.split('.').next().unwrap_or(&condition_text),
```
Comment (0xLucqs) [inline crates/cairo-lint-core/src/fix.rs:524]: what is that ?

File: crates/cairo-lint-core/tests/test_files/manual/manual_assert [hunk 3]
```diff
@@ -0,0 +1,79 @@
+//! > check-if-then
```
Comment (0xLucqs) [inline crates/cairo-lint-core/tests/test_files/manual/manual_assert:1]: add more tests for example one with `if else` don't only check the happy path you need to think of edge cases

**Review by 0xLucqs — CHANGES_REQUESTED** [review]

**Comment by 0xLucqs** [comment]
Hey i'll close this, as there's still plenty of work to do and we won't have time to do it by the end of the week. I'll still pay you though
